CONTRIBUTING.adoc

```diff
@@ -64,9 +64,37 @@ export PATH=$PATH:$GOPATH/bin
         $ git clone git://github.com/<forkid>/origin  # Replace <forkid> with the your github id
         $ cd origin
         $ git remote add upstream git://github.com/openshift/origin
-        
 
-5.  From here, you can follow https://github.com/openshift/origin/#start-developing[Start Developing] from the README.
+5.  From here, you can follow the https://github.com/openshift/origin/blob/master/HACKING.md[Hacking on OpenShift] directions, for building the release and images.  Alternatively, you can run
+
+    	 $ make clean run
+
+6.  Next, assuming you have installed Docker 1.3.2 or higher, and that you have not changed the kubernetes/openshift service subnet configuration from the default value of 172.30.0.0/16, you need to instruct the Docker daemon to trust any Docker registry on the 172.30.0.0/16 subnet.  If you are running Docker as a service via `systemd`, add the `--insecure-registry 172.30.0.0/16` argument to the options value in `/etc/sysconfig/docker` and restart the Docker daemon.  Otherwise, add "--insecure-registry 172.30.0.0/16" to the Docker daemon invocation, eg:
+
+    $ docker -d --insecure-registry 172.30.0.0/16
+
+7.  Then, the OpenShift firewalld rules are also a work in progress. For now it is easiest to disable firewalld altogether:
+
+    $ sudo systemctl stop firewalld
+
+8.  Firewalld will start again on your next reboot, but you can manually restart it with this command when you are done running OpenShift:
+
+    $ sudo systemctl start firewalld
+
+9.  Now change into the directory with the OpenShift binaries, and start the OpenShift server:
+
+    	 $ cd openshift/origin/_output/imagecontext
+	 $ sudo ./openshift start
+
+10.  Launch another terminal, change into the same directory you started OpenShift, and deploy the private docker registry within OpenShift with the certs necessary for access to master with the following commands (note, the --credentials option enables secure communication between the Docker daemon and the OpenShift server, and the --config option enables secure communication between the oadm command the OpenShift server):
```

Comment:
--credentials allows secure communication between the internal OpenShift Docker registry and the OpenShift server.

--config provides your identify (in this case, cluster-admin) to the OpenShift server.